### DISCUSSION TITLE
Time stamping for ASR

### DISCUSSION THREAD

**Discussion by Kuldeep938** [deepgram/community#601] [Feature Requests]
I have a audio file and its transcript file in text format now i want have the time stamps of the each and every word using the deepgram. How to proceed?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @Kuldeep938, Deepgram returns word-level timstamps in its JSON API response. You can find them in a list under `response["results"]["channels"][0]["alternatives"][0]["words"]`. A word will look like:
```
              {
                "word": "welcome",
                "start": 6.305,
                "end": 6.805,
                "confidence": 0.94,
                "punctuated_word": "Welcome,"
              },
```

I encourage you to also use our Playground, [playground.deepgram.com](https://playground.deepgram.com/), where you can generate sample transcripts and inspect the output structure.

**Reply by drevicko** [reply]
does this work with streamed partial responsees too?

**Reply by jkroll-deepgram** [reply]
Hi @drevicko, yes, streaming interim results also has all of this information. For example, here is an interim result, using request parameters `model=nova-2&smart_format=true&encoding=linear16&channels=1&sample_rate=16000&endpointing=true&language=en-US&interim_results=true`
```
{
    "type": "Results",
    "channel_index": [
        0,
        1
    ],
    "duration": 1.8999996,
    "start": 13.14,
    "is_final": true,
    "speech_final": true,
    "channel": {
        "alternatives": [
            {
                "transcript": "Promote the general welfare,",
                "confidence": 0.921875,
                "words": [
                    {
                        "word": "promote",
                        "start": 13.38,
                        "end": 13.88,
                        "confidence": 0.921875,
                        "punctuated_word": "Promote"
                    },
                    {
                        "word": "the",
                        "start": 13.940001,
                        "end": 14.1,
                        "confidence": 0.7001953,
                        "punctuated_word": "the"
                    },
                    {
                        "word": "general",
                        "start": 14.1,
                        "end": 14.6,
                        "confidence": 0.9941406,
                        "punctuated_word": "general"
                    },
                    {
                        "word": "welfare",
                        "start": 14.9800005,
                        "end": 15.04,
                        "confidence": 0.77978516,
                        "punctuated_word": "welfare,"
                    }
                ]
            }
        ]
    },
    "metadata": {
        "request_id": "55a9fe11-a25f-43e9-ac62-fa2944144ea1",
        "model_info": {
            "name": "2-general-nova",
            "version": "2024-01-18.26916",
            "arch": "nova-2"
        },
        "model_uuid": "c0d1a568-ce81-4fea-97e7-bd45cb1fdf3c"
    },
    "from_finalize": false
}
```
